### DISCUSSION TITLE
some questions about `variant`.

### DISCUSSION THREAD

**Discussion by hequ9** [unocss/unocss#2630] [Q&A] [answered]
i created a variant in order for `selected:bg-red` to `bg-red.active`, just like the example on the [unocss.dev](https://unocss.dev/config/variants) and it works as expected:
```typescript
variants: [
  matcher => {
    if (!matcher.startsWith('selected:'))
      return matcher
    return {
      matcher: matcher.slice(9),
      selector: s => `${ s }.active`,
    }
  },
],
```
but when I type `selected="bg-red"` it doesn't seem to work correctly, although the unocss vscode extension correctly highlights `bg-red`
```vue
<div
  bg="selected:red"
  selected="bg-red"
/>
```
![1](https://github.com/unocss/unocss/assets/49673052/e1f5aa7c-a2c7-4f1a-80ea-53622819ecdd)
![image](https://github.com/unocss/unocss/assets/49673052/f27672c4-89be-46a1-a4d4-814e997841fa)

im not sure what should i do although i noticed that #1436, so how to get the `selected` variant to work correctly as a tag attribute🙂?

**Comment by zyyv** [comment, marked as answer]
Refer to [Playground](https://unocss.dev/play/?html=AQHgJglgbsDOCmAbeBjALvMBeARBgHmgLQDuAFhBjnEqhmEQcQE6bAD0AfAFBA&config=JYWwDg9gTgLgBAbzgEwKYDNgDtUGEJaYDmANHGFKgM6owCCMMUwARgK4zDoCeZF1tAKpYIcAL5x0UCCDgByNiIDGVKnIBQ61AA9IsFBgCGbADbw0mHPkLAiACgTq45SjRhUAXHADa-N8Ig7AEo%2BV1oGJlYOLm5ggF0SJzgAN0NmQyx3L28k5zsQQxglAAtUKCC4AF4APkRc5zguODsAQgKi0qgAOioYNPcAdWAYYrs5GhNUJRhUZABaOSCKyhg2KCw4dpKy%2BucVtY3HBuPNwu2oLy3OnpNgJVQ7AE4Q3YaJqZhoLzsqCpq4AAGABIEFQxF1DNNgMlUADEidxPUxPC4Al1GIguogA&css=Q&options=N4IgLgTghgdgzgMwPYQLYgFwKgGzgUwF8g)

I would suggest write variant before css token, like `[variant]:[rules]` --> `selected-text-white`, make UnoCSS analyze it better.

**Comment by hequ9** [comment]
@zyyv very thanks for ur reply, that mean the way of `bg='[selected]:red'` is not suggested? i note the examples you provided and it has successfully solved my doubts, but i found its not work for `bg='selected:red'` and after i rewrite the `variant` all working as expected:
```typescript
matcher => {
  if (
    !matcher.startsWith('selected:')
    && !matcher.startsWith('selected-')
  )
    return matcher
  return {
    matcher: matcher.slice(9),
    selector: s => `${s}.active`,
  }
},
```
does this seem wrong and why?
